plugin/metrics/prometheus/options.go

```diff
@@ -68,11 +68,17 @@ func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
 }
 
 // InitFromViper initializes the options struct with values from Viper.
-func (opt *Options) InitFromViper(v *viper.Viper) {
+func (opt *Options) InitFromViper(v *viper.Viper) error {
 	cfg := &opt.Primary
 	cfg.ServerURL = stripWhiteSpace(v.GetString(cfg.namespace + suffixServerURL))
 	cfg.ConnectTimeout = v.GetDuration(cfg.namespace + suffixConnectTimeout)
-	cfg.TLS = cfg.getTLSFlagsConfig().InitFromViper(v)
+	if tls, err := cfg.getTLSFlagsConfig().InitFromViper(v); err == nil {
```

Comment:
The failing unit test job is due to [linter errors](https://github.com/jaegertracing/jaeger/pull/3030/checks?check_run_id=2871465877#step:5:59), the majority of which are:

> if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)

Whether if the linter is right about this being a good style or not [is debatable](https://github.com/golang/lint/issues/17#issuecomment-344050288). :)

One option is:
```
	tls, err := cfg.getTLSFlagsConfig().InitFromViper(v)
	if err != nil {
		return err
	}
	cfg.TLS = tls
	return nil
```

To check for linter errors, run `make lint`.

Reply:
@albertteoh don't think see this is the output of make lint and it does point to any of my changes 

` make lint
time staticcheck ./... \
                | grep -v \
                        -e model/model.pb.go \
                        -e proto-gen \
                        -e _test.pb.go \
                        -e thrift-gen/ \
                        -e swagger-gen/ \
                >> .lint.log || true

real    0m5.547s
user    0m5.165s
sys     0m7.180s
time gosec -quiet -exclude=G104,G107 ./...
Results:


[/Users/rkaur/github/jaeger/plugin/sampling/strategystore/adaptive/processor.go:226] - G404 (CWE-338): Use of weak random number generator (math/rand instead of crypto/rand) (Confidence: MEDIUM, Severity: HIGH)
    225: func addJitter(jitterAmount time.Duration) {
  > 226:        delay := (jitterAmount / 2) + time.Duration(rand.Int63n(int64(jitterAmount/2)))
    227:        time.Sleep(delay)



[/Users/rkaur/github/jaeger/pkg/discovery/grpcresolver/grpc_resolver.go:68] - G404 (CWE-338): Use of weak random number generator (math/rand instead of crypto/rand) (Confidence: MEDIUM, Severity: HIGH)
    67:         seed := time.Now().UnixNano()
  > 68:         random := rand.New(rand.NewSource(seed))
    69:         r := &Resolver{



[/Users/rkaur/github/jaeger/crossdock/services/tracehandler.go:308] - G404 (CWE-338): Use of weak random number generator (math/rand instead of crypto/rand) (Confidence: MEDIUM, Severity: HIGH)
    307:        // A random 8-byte hex
  > 308:        return fmt.Sprintf("%x", rand.Int63())
    309: }



[/Users/rkaur/github/jaeger/cmd/ingester/app/processor/decorator/retry.go:54] - G404 (CWE-338): Use of weak random number generator (math/rand instead of crypto/rand) (Confidence: MEDIUM, Severity: HIGH)
    53:         propagateError: false,
  > 54:         rand:           rand.New(rand.NewSource(time.Now().UnixNano())),
    55: }



[/Users/rkaur/github/jaeger/cmd/collector/app/sanitizer/cache/auto_refresh_cache.go:136] - G404 (CWE-338): Use of weak random number generator (math/rand instead of crypto/rand) (Confidence: MEDIUM, Severity: HIGH)
    135: func getRandomSleepTime(interval time.Duration) time.Duration {
  > 136:        return (interval / 2) + time.Duration(rand.Int63n(int64(interval/2)))
    137: }



[/Users/rkaur/github/jaeger/cmd/anonymizer/app/uiconv/reader.go:40] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    39: func NewReader(capturedFile string, logger *zap.Logger) (*Reader, error) {
  > 40:         cf, err := os.OpenFile(capturedFile, os.O_RDONLY, os.ModePerm)
    41:         if err != nil {



[/Users/rkaur/github/jaeger/cmd/anonymizer/app/uiconv/extractor.go:38] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    37: func NewExtractor(uiFile string, traceID string, reader *Reader, logger *zap.Logger) (*Extractor, error) {
  > 38:         f, err := os.OpenFile(uiFile, os.O_CREATE|os.O_WRONLY, os.ModePerm)
    39:         if err != nil {



[/Users/rkaur/github/jaeger/model/converter/thrift/jaeger/from_domain.go:106] - G601 (CWE-118): Implicit memory aliasing in for loop. (Confidence: MEDIUM, Severity: MEDIUM)
    105:        for idx, kv := range kvs {
  > 106:                jaegerTags[idx] = d.keyValueToTag(&kv)
    107:        }



[/Users/rkaur/github/jaeger/cmd/anonymizer/main.go:72] - G601 (CWE-118): Implicit memory aliasing in for loop. (Confidence: MEDIUM, Severity: MEDIUM)
    71:                         for _, span := range spans {
  > 72:                                 writer.WriteSpan(&span)
    73:                         }



Summary:
  Gosec  : dev
  Files  : 441
  Lines  : 73829
  Nosec  : 10
  Issues : 9

      164.77 real       166.34 user       224.53 sys
make: *** [lint-gosec] Error 1
`

Reply:
@albertteoh ooh it is `make go-lint`

Reply:
`make lint` should run `make go-lint`, but it looks like it's `gosec -quiet -exclude=G104,G107 ./...` that's failing for you before it gets to call `make go-lint`. This is odd, because `gosec` passes in CI (and for me as well).

Did you run `make install-tools` to install `gosec`? `which gosec` should point to `$GOPATH/bin/gosec`.

Reply:
installed explicitly gosec but didn't run `make install-tools`

Reply:
it should show a warning now should not show failing

Reply:
ooh less than 95 got it